rest_framework/fields.py

```diff
@@ -54,12 +54,17 @@ def is_simple_callable(obj):
         """
         True if the object is a callable that takes no arguments.
         """
-        if not callable(obj):
+        if not (inspect.isfunction(obj) or inspect.ismethod(obj)):
             return False
 
         sig = inspect.signature(obj)
         params = sig.parameters.values()
-        return all(param.default != param.empty for param in params)
+        return all(
+            param.kind == param.VAR_POSITIONAL or
+            param.kind == param.VAR_KEYWORD or
+            param.default != param.empty
+            for param in params
+        )
```

Comment:
@rpkilby: Can you just explain this call to me, just for my sanity? 🙂

I'm expecting _any(iterable)_ — it looks like _bool or bool or generator_, so why isn't it this:

```
>>> a = True
>>> b = True
>>> def c():
...     yield True
...     yield False
>>> all(a or b or c())
TypeError: 'bool' object is not iterable
```

Sorry for being slow. Thanks!